toml/src/main/kotlin/org/rust/toml/inspections/CrateNotFoundInspection.kt

```diff
@@ -0,0 +1,84 @@
+import org.toml.lang.psi.ext.TomlLiteralKind
+import org.toml.lang.psi.ext.kind
+
+class CrateNotFoundInspection : LocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        buildVisitor(holder) ?: super.buildVisitor(holder, isOnTheFly)
+
+    private fun buildVisitor(holder: ProblemsHolder): PsiElementVisitor? {
+        if (!isFeatureEnabled(RsExperiments.CRATES_LOCAL_INDEX)) return null
+        if (holder.file.name != CargoConstants.MANIFEST_FILE) return null
+
+        return object : TomlVisitor() {
+            override fun visitKeyValue(element: TomlKeyValue) {
+                val table = element.parent as? TomlTable ?: return
+                val depTable = getDependencyTable(table) ?: return
+                if (depTable !is DependencyTableBase.DependencyTable) return
+
+                val segment = element.key.segments.getOrNull(0) ?: return
+                val name = segment.name ?: return
+                val value = element.value ?: return
+                if (value is TomlLiteral && value.kind is TomlLiteralKind.String) {
+                    handleDependency(name, segment, holder)
+                } else if (value is TomlInlineTable) {
+                    handleDependency(name, segment, holder)
+                }
+            }
+
+            override fun visitTable(element: TomlTable) {
+                val depTable = getDependencyTable(element) ?: return
+                if (depTable !is DependencyTableBase.SpecificDependencyTable) return
+
+                handleDependency(depTable.crateName, depTable.crate, holder)
+            }
+        }
+    }
+
+    private fun handleDependency(crateName: String, nameElement: TomlKeySegment, holder: ProblemsHolder) {
+        if (CratesLocalIndexService.getInstance().getCrate(crateName) == null) {
+            holder.registerProblem(nameElement, "Crate $crateName not found")
+        }
+    }
+}
+
+private fun getDependencyTable(table: TomlTable): DependencyTableBase? {
+    val key = table.header.key ?: return null
+    val segments = key.segments
+    val dependencyNameIndex = segments.indexOfFirst { it.isDependencyKey }
+
+    return when {
+        // [dependencies], [x86.dev-dependencies], etc.
+        dependencyNameIndex == segments.lastIndex -> DependencyTableBase.DependencyTable
+        // [dependencies.crate]
+        dependencyNameIndex != -1 -> {
+            val crate = segments.getOrNull(dependencyNameIndex + 1)
+            val crateName = crate?.name
+            if (crate != null && crateName != null) {
+                DependencyTableBase.SpecificDependencyTable(crate, crateName)
+            } else {
+                null
+            }
+        }
+        else -> null
+    }
+}
+
+private sealed class DependencyTableBase {
+    object DependencyTable : DependencyTableBase()
+    data class SpecificDependencyTable(val crate: TomlKeySegment, val crateName: String) : DependencyTableBase()
+}
```

Comment:
What do you think about this naming and fields order?
```suggestion
private sealed class DependencyTable {
    object General : DependencyTable()
    data class Specific(val crateName: String, val crateElement: TomlKeySegment) : DependencyTable()
}
```